src/Home.jsx

```diff
@@ -0,0 +1,93 @@
+import {useState} from "react";
+import Country from "./components/Country.jsx";
+import Header from "./components/Header.jsx";
+import RegionFilter from "./components/RegionFilter.jsx";
+import SearchBox from "./components/SearchBox.jsx";
+import countriesData from './assets/json/CountriesData.json';
+
+/**
+ * Home Component
+ * Manages the main layout, filtering, and display of countries.
+ *
+ * Props:
+ * - theme: Current theme ("light" or "dark").
+ * - toggleTheme: Function to toggle the theme.
+ *
+ * Functionality:
+ * - Displays a list of countries with search and region filtering.
+ * - Allows toggling between light and dark themes.
+ */
+const Home = ({theme, toggleTheme}) => {
+    const [countries, setCountries] = useState(countriesData);
+
+    /**
+     * Filters the countries list to display details of a single country by name.
+     */
+    const showOneCountryDetails = (countryName) => {
+        setCountries(countriesData.filter((country) => country.name === countryName));
+    };
+
+    /**
+     * Resets the countries list to show all available countries.
+     */
+    const showAllCountries = () => {
+        setCountries(countriesData);
```

Comment:
It is not a good practice too add comment that explain what a function of a variable is doing,  the code should explain itself